addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -29,6 +29,35 @@ class StockPickingBatch(models.Model):
         help="Priority of a batch is the maximum priority of its pickings."
     )
 
+    state = fields.Selection(
+        selection_add=[
+            ('waiting', 'Waiting'),
+            ('ready', 'Ready'),
+        ],
+        compute='_compute_state',
+        store=True,
+    )
+
+    @api.multi
+    def confirm_picking(self):
+        """Over write method to use our waiting state and recompute state
+           incase this should be ready or in_progress
+        """
+        pickings_todo = self.mapped('picking_ids')
+        self.write({'state': 'waiting'})  # Get it out of draft
+        res = pickings_todo.action_assign()
+        self._compute_state()
```

Comment:
And this will trigger the automatic transition to `ready`, in case all pickings are `ready`, right?